packages/@aws-cdk/aws-events-targets/test/ecs/event-rule-target.test.ts

```diff
@@ -647,3 +647,208 @@ test('uses the specific fargate platform version', () => {
+        EcsParameters: {
+          TaskCount: 1,
+          TaskDefinitionArn: { Ref: 'TaskDef54694570' },
+          LaunchType: 'FARGATE',
+          NetworkConfiguration: {
+            AwsVpcConfiguration: {
+              Subnets: [
+                {
+                  Ref: 'Vpc2PublicSubnet1Subnet758D49A9',
+                },
+              ],
+              AssignPublicIp: 'ENABLED',
+              SecurityGroups: [
+                {
+                  'Fn::GetAtt': [
+                    'TaskDefSecurityGroupD50E7CF0',
+                    'GroupId',
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        Input: '{"containerOverrides":[{"name":"TheContainer","command":["echo","yay"]}]}',
+        RoleArn: { 'Fn::GetAtt': ['TaskDefEventsRoleFB3B67B8', 'Arn'] },
+        Id: 'Target0',
+      },
+    ],
+  });
+});
+
+test('With assignPublicIp=false and public subnet AssignPublicIp=DISABLED', () => {
+  // GIVEN
+  vpc = new ec2.Vpc(stack, 'Vpc2', {
+    maxAzs: 1,
+    subnetConfiguration: [{
+      subnetType: ec2.SubnetType.PUBLIC,
+      name: 'Public',
+    }],
+  });
+  cluster = new ecs.Cluster(stack, 'EcsCluster2', { vpc });
+
+  const taskDefinition = new ecs.FargateTaskDefinition(stack, 'TaskDef');
+  taskDefinition.addContainer('TheContainer', {
+    image: ecs.ContainerImage.fromRegistry('henk'),
+  });
+
+  const rule = new events.Rule(stack, 'Rule', {
+    schedule: events.Schedule.expression('rate(1 min)'),
+  });
+
+  // WHEN
+  rule.addTarget(new targets.EcsTask({
+    assignPublicIp: false,
+    cluster,
+    taskDefinition,
+    taskCount: 1,
+    subnetSelection: { subnetType: ec2.SubnetType.PUBLIC },
+    containerOverrides: [{
+      containerName: 'TheContainer',
+      command: ['echo', 'yay'],
+    }],
+  }));
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::Events::Rule', {
+    Targets: [
+      {
+        Arn: { 'Fn::GetAtt': ['EcsCluster2F191ADEC', 'Arn'] },
+        EcsParameters: {
+          TaskCount: 1,
+          TaskDefinitionArn: { Ref: 'TaskDef54694570' },
+          LaunchType: 'FARGATE',
+          NetworkConfiguration: {
+            AwsVpcConfiguration: {
+              Subnets: [
+                {
+                  Ref: 'Vpc2PublicSubnet1Subnet758D49A9',
+                },
+              ],
+              AssignPublicIp: 'DISABLED',
+              SecurityGroups: [
+                {
+                  'Fn::GetAtt': [
+                    'TaskDefSecurityGroupD50E7CF0',
+                    'GroupId',
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        Input: '{"containerOverrides":[{"name":"TheContainer","command":["echo","yay"]}]}',
+        RoleArn: { 'Fn::GetAtt': ['TaskDefEventsRoleFB3B67B8', 'Arn'] },
+        Id: 'Target0',
+      },
+    ],
+  });
+});
+
+test('With assignPublicIp=true and isolated subnet AssignPublicIp=ENABLED', () => {
```

Comment:
I don't know if we should allow this, but I used this test case to illustrate the question.